aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/DeleteHandler.java

```diff
@@ -49,7 +49,7 @@ protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
         // https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-deletionpolicy.html
         // For AWS::RDS::DBInstance resources that don't specify the DBClusterIdentifier property, the default policy is Snapshot.
         // Final snapshots are not allowed for read replicas and cluster instances.
-        if (BooleanUtils.isTrue(request.getSnapshotRequested()) &&
+        if (BooleanUtils.isNotFalse(request.getSnapshotRequested()) &&
```

Comment:
Just to put on radar. This will make contract tests leave snapshot behind whenever they run into any AWS account. THese snapshot will sum up till quota is full ~50 snapshots then contract test will start to fail. Unfortunately there is no way around to write the parameter in contract test request.

Reply:
Ack, many thanks for raising it. We will solve it progressively. I believe we can justify a temporary cleanup TOIL by loosing a risk of an accidental snapshot removal.